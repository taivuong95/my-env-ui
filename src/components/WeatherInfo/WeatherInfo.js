import React from 'react';

const WeatherInfo = () => {
  return (
    <div data-test="weather-info-layout" className='weather-info-layout'>
      <div className='weather-info__item'>
        <div className='weather-info__item--title'>PSI</div>
        <div className='weather-info__item--number'>23</div>
        <div className='weather-info__item--review'>Good</div>
      </div>
      <div className='weather-info__item'>
        <div className='weather-info__item--title'>RAIN</div>
        <div className='weather-info__item--number-nobg'>0</div>
        <div className='weather-info__item--review'>mm</div>
      </div>
      <div className='weather-info__item'>
        <div className='weather-info__item--title'>DENGUE</div>
        <div className='weather-info__item--circle'></div>
        <div className='weather-info__item--review'></div>
      </div>
      <div className='weather-info__item'>
        <div className='weather-info__add-btn'></div>
        <div className='weather-info__add-txt'>Add</div>
      </div>
    </div>
  )
}

export default WeatherInfo;