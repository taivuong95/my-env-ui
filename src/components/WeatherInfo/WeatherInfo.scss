.weather-info-layout{
    display: flex;
  }
  .weather-info{
    &__item{
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      align-items: center;
      width: 100%;
      height: 12rem;
      padding: 1rem 0;
      &--title{
        font-weight: 500;
        font-size: 1.8rem;
      }
      &--number{
        font-size: 2.4rem;
        padding: 8px 8px 8px 15px;
        background: #32a848;
        border-radius: 4px;
        width: 6rem;
        height: 5rem;
      }
      &--number-nobg{
        font-size: 3rem;
      }
      &--review{
        font-size: 1.6rem;
      }
      &--circle{
        width: 4rem;
        height: 4rem;
        border: 2px solid #fff;
        border-radius: 100%;
        background-color: #94d6f7;
      }
      &:not(:last-of-type){
        border-right: 2px solid #fff;
      }
      &:last-child{
        justify-content: space-around;
      }
    }
    &__add-btn{
      display: flex;
      justify-content: center;
      align-items: center;
      width: 30px;
      height: 30px;
      border: 1px solid #333;
      border-radius: 100%;
      text-align: center;
      cursor: pointer;

      &::before{
        display: inline-block;
        width: 20px;
        height: 20px;
        background-image: url(../../assets/img/add.svg);
        background-size: contain;
        content: "";
      }
    }
    &__add-txt{
      color: #333;
      font-weight: 500;
      font-size: 20px;
    }
  }
  
