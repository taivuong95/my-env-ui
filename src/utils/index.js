export const KelvinToCelsius = (kelvin) => {
    return kelvin - 273.15;
}