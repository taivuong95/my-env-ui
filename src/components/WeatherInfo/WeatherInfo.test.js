import React from 'react';
import WeatherInfo from './WeatherInfo';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr } from '../../utils/index';

const setup = () => shallow(<WeatherInfo />);
describe('WeatherInfo Component', () => {
    test("expect to render WeatherInfo component", () => {
        const wrapper = setup();
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const wrapper = setup();
        const navComponent = findByTestAttr(wrapper, "weather-info-layout");
        expect(navComponent.length).toBe(1);
    })
})