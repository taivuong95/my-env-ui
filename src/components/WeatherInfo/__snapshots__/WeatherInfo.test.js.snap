// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo Component expect to render WeatherInfo component 1`] = `
<div
  className="weather-info-layout"
  data-test="weather-info-layout"
>
  <div
    className="weather-info__item"
  >
    <div
      className="weather-info__item--title"
    >
      PSI
    </div>
    <div
      className="weather-info__item--number"
    >
      23
    </div>
    <div
      className="weather-info__item--review"
    >
      Good
    </div>
  </div>
  <div
    className="weather-info__item"
  >
    <div
      className="weather-info__item--title"
    >
      RAIN
    </div>
    <div
      className="weather-info__item--number-nobg"
    >
      0
    </div>
    <div
      className="weather-info__item--review"
    >
      mm
    </div>
  </div>
  <div
    className="weather-info__item"
  >
    <div
      className="weather-info__item--title"
    >
      DENGUE
    </div>
    <div
      className="weather-info__item--circle"
    />
    <div
      className="weather-info__item--review"
    />
  </div>
  <div
    className="weather-info__item"
  >
    <div
      className="weather-info__add-btn"
    />
    <div
      className="weather-info__add-txt"
    >
      Add
    </div>
  </div>
</div>
`;
